src/clipboard_handlers/tables_clipboard.ts

```diff
@@ -166,16 +179,20 @@ export class TableClipboardHandler extends AbstractCellClipboardHandler<
     sheetId: UID,
     tableCell: TableCell,
     position: CellPosition,
+    nRows: number,
+    nCols: number,
     options?: ClipboardOptions
   ) {
     if (tableCell.table && !options?.pasteOption) {
-      const { range: tableRange } = tableCell.table;
-      const zoneDims = zoneToDimension(this.getters.getRangeFromRangeData(tableRange).zone);
+      const { range: tableRange, dimension } = tableCell.table;
+      const zone = this.getters.getRangeFromRangeData(tableRange).zone;
+      const zoneDims = zoneToDimension(zone);
+      const { numberOfCols, numberOfRows } = dimension || zoneDims;
       const newTableZone = {
         left: position.col,
         top: position.row,
-        right: position.col + zoneDims.numberOfCols - 1,
-        bottom: position.row + zoneDims.numberOfRows - 1,
+        right: position.col + Math.min(numberOfCols, nCols) - 1,
+        bottom: position.row + Math.min(numberOfRows, nRows) - 1,
```

Comment:
In what cases using tableCell.table.dimensions isn't enough ? Why do we also need to check the size of the original table zone and the number of tableCells ?

Reply:
You're right. It was needed before the last version, when I wasn't recomputing `tableCell.table.dimensions` but using the original table zone.